tracking/src/main/java/org/hps/recon/tracking/kalman/StateVector.java

```diff
@@ -337,37 +344,51 @@ StateVector smooth(StateVector snS, StateVector snP) {
                     snS.kLow, snS.kUp, snP.kLow, snP.kUp);
         StateVector sS = this.copy();
 
-        // solver.setA defines the input matrix and checks whether it is singular. A copy is needed because the input gets modified.
+        // solver.setA defines the input matrix and checks whether it is singular. 
+        // A copy is needed because the input gets modified.
         if (!solver.setA(snP.helix.C.copy())) {
-            logger.fine("StateVector:smooth, inversion of the covariance matrix failed");
-            //snP.helix.C.print();
-            //SquareMatrix invrs = KalTrack.mToS(snP.helix.C).invert();
-            //invrs.print("inverse");
-            //invrs.multiply(KalTrack.mToS(snP.helix.C)).print("unit matrix?");            
-            for (int i=0; i<5; ++i) {      // Fill the inverse with something not too crazy and continue . . .
-                for (int j=0; j<5; ++j) {
-                    if (i == j) {
-                        Cinv.unsafe_set(i,j,1.0/Cinv.unsafe_get(i,j));
-                    } else {
-                        Cinv.unsafe_set(i, j, 0.); 
+            SquareMatrix invrs = KalTrack.mToS(snP.helix.C).fastInvert();
```

Comment:
For future, this can easily be turned into a stream with a lambda instead. Don't think it's needed now, but wanted to note it.

Reply:
This applies to most for loops ...